distribution/src/assembly/bin.xml

```diff
@@ -63,7 +63,7 @@
 
     <files>
         <file>
-            <source>target/WSO2Carbon/conf/default/deployment.yaml</source>
+            <source>carbon-home/conf/deployment.yaml</source>

```

Comment:
Since deployment.yaml is runtime specific as mentioned in README of the kernel feature, therefore the deployment.yaml need to be copied to the runtime specific in the conf folder